mm2src/mm2_main/tests/docker_tests/docker_tests_inner.rs

```diff
@@ -2598,6 +2598,107 @@ fn test_get_max_taker_vol_with_kmd() {
     block_on(mm_alice.stop()).unwrap();
 }
 
+#[test]
+fn test_get_max_maker_vol() {
+    let (_ctx, _, priv_key) = generate_utxo_coin_with_random_privkey("MYCOIN1", 1.into());
+    let coins = json!([mycoin_conf(1000), mycoin1_conf(1000)]);
+    let mm = MarketMakerIt::start(
+        json!({
+            "gui": "nogui",
+            "netid": 9000,
+            "dht": "on",  // Enable DHT without delay.
+            "passphrase": format!("0x{}", hex::encode(priv_key)),
+            "coins": coins,
+            "rpc_password": "pass",
+            "i_am_seed": true,
+        }),
+        "pass".to_string(),
+        None,
+    )
+    .unwrap();
+    let (_dump_log, _dump_dashboard) = mm_dump(&mm.log_path);
+
+    log!("{:?}", block_on(enable_native(&mm, "MYCOIN", &[])));
+    log!("{:?}", block_on(enable_native(&mm, "MYCOIN1", &[])));
+    let rc = block_on(mm.rpc(&json!({
+        "userpass": mm.userpass,
+        "mmrpc": "2.0",
+        "method": "max_maker_vol",
+        "params": {
+            "coin": "MYCOIN1",
+        }
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!max_taker_vol: {}", rc.1);
+
+    // 1 - tx_fee
+    let expected_volume = MmNumber::from("0.99999");
+    let expected = MaxMakerVolResponse {
+        coin: "MYCOIN1".to_string(),
+        volume: MmNumberMultiRepr::from(expected_volume.clone()),
+    };
+    let actual: RpcSuccessResponse<MaxMakerVolResponse> = serde_json::from_str(&rc.1).unwrap();
+    assert_eq!(actual.result, expected);
+
+    let rc = block_on(mm.rpc(&json!({
+        "userpass": mm.userpass,
+        "method": "setprice",
+        "base": "MYCOIN1",
+        "rel": "MYCOIN",
+        "price": 1,
+        "max": true,
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!setprice: {}", rc.1);
+    let res: SetPriceResponse = serde_json::from_str(&rc.1).unwrap();
```

Comment:
There is `set_price` helper in `for_tests` that can be also refactored to return `SetPriceResponse`.